apps/system/test/unit/lockscreen/widgets/clock/lockscreen_clock_widget_test.js

```diff
@@ -49,4 +49,50 @@ suite('LockScreenClockWidget > ', function() {
       mockThis.resources.elements.date.textContent,
       `it doesn't update the date.textContent with the date in locale format`);
   });
+
+  test(`it would update the alarm info (an alarm is set)`, function() {
```

Comment:
For async test with Promise, you need to write it as:

    test(`....`, function(done) {
        promise....then(/* do something */)
        .then(() => {  assert.isTrue();  }    // assert in one then step
        .then(done)    // the `done` will receive nothing and the runner will know it ends without any error
        .catch(done)  // the `done` will receive an error object and let the test runner know there is an error
    })